### PR TITLE
Improve the SQL version comparisons:

### LINKED ISSUE #466: GET /api/v1/package/{uuid} fails with some packages
Steps to reproduce:
- Start a clean instance of trustify
- Enable only the "redhat-csaf" importer. Wait some minutes
- Search for packages like ose-powervs-cloud-controller-manager-rhel8 and try to fetch it by their uuid, we will see an error:

```json
{
    "error": "Database error",
    "message": "Query Error: error returned from database: invalid input syntax for type numeric: \"sha256:f579ee538e6f4bfa4d9c20bf2680d7fc3ec7a757a40637a743b7962187b4d6a1\""
}
```

https://github.com/trustification/trustify/assets/2582866/6e1012b5-f0e1-43a8-b1d0-9eb1284b4044

> Hint: it seems to happen with package whose version are something like sha256:cfb31631b19c2e9d2ef28bac6d4471c97e79976b0c9590032cf0206bfcd3e4f3 ; on the other hand, packages with versions like 1.2.3 work fine

### PR DESCRIPTION
**Pull request by bobmcwhirter**
- use bigger ints for jackholes who version stuff like 202410298675309.
- if a version cmp fails (because a sha:xxx "version" was given) don't fail, just false.

In theory fixes #466.

### CODE REVIEW

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        return 0;
+    end if;
+
+    return null;
+exception
+    when others then
+        return null;
+end
+$$
+    language 'plpgsql';
```
Comment (JimFuller-RedHat) [inline migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:128]: have u considered adding IMMUTABLE PARALLEL SAFE; here ?

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    right_build text;
+
+    left_numeric bool;
+    right_numeric bool;
+
+    cur integer;
+
+begin
+
+    raise notice 'semver_cmp % %', left_p, right_p;
```
Comment (JimFuller-RedHat) [inline migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:29]: FWIW - **log_min_messages** configuration parameter to a higher level, such as WARNING or ERROR, to suppress NOTICE messages ... useful for production when we dont want to generate gigantic log files ;)

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql [hunk 3]
```diff
@@ -0,0 +1,128 @@
+
+create or replace function semver_cmp(left_p text, right_p text)
```
Comment (JimFuller-RedHat) [inline migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:2]: it is unclear if this function covers off all of http://ftp.rpm.org/api/4.4.2.2/rpmvercmp_8c-source.html .... I did a version in sql here https://github.com/RedHatProductSecurity/component-registry/blob/main/corgi/core/migrations/0092_install_stored_proc.py ... if I have the time tmrw will cross reference between rpmvercmp and this

**Review by JimFuller-RedHat — COMMENTED** [review]
I would test output of  this cmp with rpmvercmp ..

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 3 above)
Comment (JimFuller-RedHat) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:2]: and yes I know this comment is expanding scope ;)

**Review by JimFuller-RedHat — APPROVED** [review]

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 1 above)
Comment (bobmcwhirter) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:128]: Considering that I've no idea what that means... no. Should I?

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 2 above)
Comment (bobmcwhirter) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:29]: fwiw, I can never actually find the postgres logs. raise notice shows up in the rust log if `trace` at least.

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 3 above)
Comment (bobmcwhirter) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:2]: So far only semver cmp. Need to do *many more*.

Comment (bobmcwhirter) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:2]: And I'll definitely steal your RPM version cmp.

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 1 above)
Comment (JimFuller-RedHat) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:128]: as this function does no mutation, and only works on 2 supplied inputs ... I would suggest it ... usually results in much faster function.

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 2 above)
Comment (JimFuller-RedHat) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:29]: ya in production, you might find this jams up logs unnec ... just mentioned

File: migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql (hunk 3 above)
Comment (JimFuller-RedHat) [reply migration/src/m0000475_improve_version_comparison_fns/semver_cmp.sql:2]: +1 - we assumed rpm folks have encountered every variation and stole from them thusly ... I think my sql version is correct eg. ran it against their test cases (but did not embed in corgi)

**Pull request merged**